### ISSUE TITLE
Reduce with fp32 dest acc enabled fails

### ISSUE THREAD

**Issue by bbradelTT** [tenstorrent/tt-metal#15744]
While working on improving accuracy for reduce, testing on BH failed.

https://github.com/tenstorrent/tt-metal/actions/runs/12160963233/job/33915026914
```
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/.ttnn_runtime_artifacts/tt_metal/third_party/tt_llk_blackhole/llk_lib/llk_math_reduce.h:195:23: error: static assertion failed
  195 |         static_assert(!is_fp32_dest_acc_en);
      |                       ^~~~~~~~~~~~~~~~~~~~
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/.ttnn_runtime_artifacts/tt_metal/third_party/tt_llk_blackhole/llk_lib/llk_math_reduce.h:195:23: note: 'false' evaluates to false
```

fp32 dest acc is needed for better accuracy, and needs to work on BH without any asserts being thrown.

Steps to reproduce:
https://github.com/tenstorrent/tt-metal/compare/main...refs/heads/bbradel-13621_fp32
then run failing part of BH post commit. Specifically
```
pytest tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py::test_run_reduce_max_hw_test[min-input_shapes0]
```

Expected:
test passes

Actual:
test fails

**Comment by ncvetkovicTT** [comment]
Passing with [PR#54](https://github.com/tenstorrent/tt-llk-bh/pull/54). The fix will be a part of [PR#15754](https://github.com/tenstorrent/tt-metal/pull/15754)

```
2024-12-05 18:22:21.089 | WARNING  | SiliconDriver   - Unknown board type for chip 0. This might happen because chip is running old firmware. Defaulting to UNKNOWN
2024-12-05 18:22:21.089 | WARNING  | SiliconDriver   - Unknown board type for chip 0. This might happen because chip is running old firmware. Defaulting to UNKNOWN
2024-12-05 18:22:21.089 | INFO     | SiliconDriver   - Detected 1 PCI device : [0]
2024-12-05 18:22:21.089 | INFO     | SiliconDriver   - Opened PCI device 0; KMD version: 1.29.0, IOMMU: disabled
                  Metal | INFO     | Initializing device 0. Program cache is NOT enabled
                 Device | INFO     | For Blackhole hardcode AICLK to 800 MHz due to lack of ARC message support
                  Metal | INFO     | AI CLK for device 0 is:   800 MHz
2024-12-05 18:22:22.305 | INFO     | tests.tt_eager.python_api_testing.sweep_tests.run_pytorch_ci_tests:run_single_pytorch_test:68 - Running with shape: [[1, 1, 32, 32]] on device: 0
2024-12-05 18:22:22.305 | DEBUG    | tests.tt_eager.python_api_testing.sweep_tests.common:run_tt_lib_test:43 - Running with args: {'dtype': [<DataType.BFLOAT16: 0>], 'layout': [<Layout.TILE: 1>], 'input_mem_config': [MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt)], 'output_mem_config': MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt)}
2024-12-05 18:22:22.727 | INFO     | tests.tt_eager.python_api_testing.sweep_tests.common:run_tt_lib_test:56 - True Max ATOL Delta: 0.0, Max RTOL Delta: 0.0, PCC: 1.0
2024-12-05 18:22:22.727 | DEBUG    | tests.tt_eager.python_api_testing.sweep_tests.run_pytorch_ci_tests:run_single_pytorch_test:79 - Test pass/fail: True with Max ATOL Delta: 0.0, Max RTOL Delta: 0.0, PCC: 1.0
2024-12-05 18:22:22.727 | DEBUG    | tests.tt_eager.python_api_testing.sweep_tests.run_pytorch_ci_tests:run_single_pytorch_test:80 - Test args: {'dtype': [<DataType.BFLOAT16: 0>], 'layout': [<Layout.TILE: 1>], 'input_mem_config': [MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt)], 'output_mem_config': MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt)}
2024-12-05 18:22:22.727 | INFO     | tests.tt_eager.python_api_testing.sweep_tests.run_pytorch_ci_tests:run_single_pytorch_test:89 - reduce-min-hw test passed with input shape [[1, 1, 32, 32]].
PASSED                  Metal | INFO     | Closing device 0
                  Metal | INFO     | Disabling and clearing program cache on device 0


=============================================================================================================================================================================================================== warnings summary ===============================================================================================================================================================================================================
tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py::test_run_reduce_max_hw_test[min-input_shapes0]
  tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py:77: PytestWarning: record_property is incompatible with junit_family 'xunit2' (use 'legacy' or 'xunit1')
    @pytest.mark.parametrize(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================================================================================================================================== PASSES ====================================================================================================================================================================================================================
------------------------------------------------------------------------------------------------------------------------------------------------------------------------ generated xml file: /localdev/ncvetkovic/tt-metal/generated/test_reports/most_recent_tests.xml ------------------------------------------------------------------------------------------------------------------------------------------------------------------------
============================================================================================================================================================================================================== slowest durations ===============================================================================================================================================================================================================
1.22s setup    tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py::test_run_reduce_max_hw_test[min-input_shapes0]
0.42s call     tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py::test_run_reduce_max_hw_test[min-input_shapes0]
0.00s teardown tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py::test_run_reduce_max_hw_test[min-input_shapes0]
=========================================================================================================================================================================================================== short test summary info ============================================================================================================================================================================================================
PASSED tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_reduce.py::test_run_reduce_max_hw_test[min-input_shapes0]
========================================================================================================================================================================================================= 1 passed, 1 warning in 1.74s =========================================================================================================================================================================================================
                 Device | INFO     | Closing user mode device drivers
```

**Comment by ncvetkovicTT** [comment]
PR #15754 is merged, the issue can be closed.
